src/os/bluestore/BlueStore.cc

```diff
@@ -498,6 +505,41 @@ static bool is_extent_shard_key(const string& key)
   return *key.rbegin() == EXTENT_SHARD_KEY_SUFFIX;
 }
 
+// offloaded blob keys are the onode key, plus a u32, plus 'B'.  the trailing
+// char lets us quickly test whether it is a blob key without decoding any
+// of the prefix bytes.
```

Comment:
I think this is the big question for me: do we want to put this data "inline" with the onode metadata, which will give us prefecthing/locality for this one object but make it harder to keep all onode ssts in the block cache, or do we put it in a different part of the namespace with minimal locality (only temporal), which gets us efficient storage but no real prefetching behavior (there will still be an io).

Reply:
I don't have much intuition as to which will perform better in the end. I suspect we'll have to test both (with a full device) to know the answer.

Reply:
IMO "inline" approach is less flexible. E.g. it hardly allows "sparse" object mode - i.e. object consisting of many tiny blobs. You already mentioned negative impact on cache utilization.
And I think additional I/O to retrieve payload data isn't a big deal - we do that in regular operations anyway. Moreover this patch mainly about more efficient space utilization rather than read op performance improvement. I think we shouldn't care about small object reading performance much...